### ISSUE TITLE
build issue on OS X 10.11: lippicv not found

### ISSUE THREAD

**Issue by MichaMucha** [dmlc/mxnet#1064]
I tried all sorts of solutions and cannot get past this. What am I doing wrong?

$ cp make/osx.mk ./config.mk; make
g++ -DMSHADOW_FORCE_STREAM -Wall -O3 -I./mshadow/ -I./dmlc-core/include -fPIC -Iinclude -msse3 -funroll-loops -Wno-unused-parameter -Wno-unknown-pragmas -I/usr/local/cuda/include -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -I/System/Library/Frameworks/Accelerate.framework/Versions/Current/Frameworks/vecLib.framework/Versions/Current/Headers/ -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMXNET_USE_OPENCV=1 `pkg-config --cflags opencv` -fopenmp  -shared -o lib/libmxnet.so build/resource.o build/c_api/c_api.o build/c_api/c_api_error.o build/c_api/c_predict_api.o build/common/mxrtc.o build/common/tblob_op_registry.o build/engine/engine.o build/engine/naive_engine.o build/engine/threaded_engine.o build/engine/threaded_engine_perdevice.o build/engine/threaded_engine_pooled.o build/io/io.o build/io/iter_csv.o build/io/iter_image_recordio.o build/io/iter_mnist.o build/kvstore/kvstore.o build/ndarray/ndarray.o build/ndarray/ndarray_function.o build/ndarray/unary_function.o build/operator/activation.o build/operator/batch_norm.o build/operator/block_grad.o build/operator/concat.o build/operator/convolution.o build/operator/cross_device_copy.o build/operator/cudnn_batch_norm.o build/operator/deconvolution.o build/operator/dropout.o build/operator/elementwise_binary_op.o build/operator/elementwise_binary_scalar_op.o build/operator/elementwise_sum.o build/operator/embedding.o build/operator/fully_connected.o build/operator/identity_attach_KL_sparse_reg.o build/operator/leaky_relu.o build/operator/lrn.o build/operator/native_op.o build/operator/ndarray_op.o build/operator/operator.o build/operator/pooling.o build/operator/regression_output.o build/operator/reshape.o build/operator/slice_channel.o build/operator/softmax_activation.o build/operator/softmax_output.o build/operator/swapaxis.o build/operator/upsampling.o build/optimizer/optimizer.o build/optimizer/sgd.o build/storage/storage.o build/symbol/graph_executor.o build/symbol/static_graph.o build/symbol/symbol.o dmlc-core/libdmlc.a build/ndarray/ndarray_function_gpu.o build/ndarray/unary_function_gpu.o build/operator/activation_gpu.o build/operator/batch_norm_gpu.o build/operator/block_grad_gpu.o build/operator/concat_gpu.o build/operator/convolution_gpu.o build/operator/cudnn_batch_norm_gpu.o build/operator/deconvolution_gpu.o build/operator/dropout_gpu.o build/operator/elementwise_binary_op_gpu.o build/operator/elementwise_binary_scalar_op_gpu.o build/operator/elementwise_sum_gpu.o build/operator/embedding_gpu.o build/operator/fully_connected_gpu.o build/operator/identity_attach_KL_sparse_reg_gpu.o build/operator/leaky_relu_gpu.o build/operator/lrn_gpu.o build/operator/native_op_gpu.o build/operator/pooling_gpu.o build/operator/regression_output_gpu.o build/operator/reshape_gpu.o build/operator/slice_channel_gpu.o build/operator/softmax_activation_gpu.o build/operator/softmax_output_gpu.o build/operator/swapaxis_gpu.o build/operator/upsampling_gpu.o build/optimizer/sgd_gpu.o -pthread -lm -lcudart -lcublas -lcurand -L/usr/local/cuda/lib64 -L/usr/local/cuda/lib -framework Accelerate  `pkg-config --libs opencv`  -lnvrtc -lcuda
clang: warning: argument unused during compilation: '-pthread'
ld: warning: directory not found for option '-L/usr/local/cuda/lib64'
ld: warning: directory not found for option '-L/install/prefix/lib'
ld: warning: path '/usr/local/lib/libomp.dylib' following -L not a directory
ld: library not found for -lippicv
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [lib/libmxnet.so] Error 1

**Comment by piiswrong** [comment]
Where is your cuda installed?

**Comment by MichaMucha** [comment]
/usr/local/cuda

**Comment by MichaMucha** [comment]
I found this:
http://sourceforge.net/projects/opencvlibrary/files/3rdparty/ippicv/

however I don't know how to do use these files. I tried adding the full path to it in ADD_LDFLAGS in the config.mk file (actually osx.mk which I run). The error then changed to 
`ld: library not found for -lfullpath/ippicv`
so essentially nothing changed

**Comment by piiswrong** [comment]
I'm guessing your opencv is not properly installed. Try setting USE_OPENCV to 0 in your osx.mk. If you want to use opencv, you need to install it with brew, or if that doesn't work, from source.

**Comment by MichaMucha** [comment]
It is installed from brew, with the flag --with-contrib, could that be the issue? that this here causes the problem?

Do I understand it correctly that the mxnet build throws error expecting me to have Intel BLAS installed?

**Comment by MichaMucha** [comment]
Okay thanks for spending your time on this. I figured it out, turned out to be very simple.
I put the contents of http://sourceforge.net/projects/opencvlibrary/files/3rdparty/ippicv/
into /usr/local/lib/

while i should have put it to /usr/local/ , letting what was in include/ and in lib/ fall into the right folder.
No problems after that.
Thanks

**Comment by erickingxu** [comment]
Hi  @MichaMucha ,I had the same problem with "brew install opencv" command line.
Can you make it clear.
1.the url is lost ,you mean that i need to copy ippicv into /usr/local/lib/ ?
but i found it still can not work with the problem
thx

**Comment by erickingxu** [comment]
Thx @piiswrong and @MichaMucha ,
I found a reslution for this problem just with two command lines in osx terminal:

locate ippicv
sudo ln -s /usr/local/share/OpenCV/3rdparty/lib/libippicv.a /usr/local/lib/libippicv.a
